### PR TITLE
More cleaning

### PR DESCRIPTION
**Pull request by kristian-clausal**
Fixing annotations, mostly.

I've looked a bit at mypy output for the other extractors, and it's helped to find issues with annotations in wikitextprocessor and elsewhere in wiktextract (for example adding some `@overload` stuff). @xxyzz do you mind if I start poking around in there? It should be non-destructive (hopefully).

### CODE REVIEW

**Comment by kristian-clausal** [comment]
Leaving this uncommitted until next week, not a high priority.

**Comment by xxyzz** [comment]
I haven't use mypy for a while, I think pydantic and ruff gives more useful suggestions. Maybe mypy could find more meaningful errors in code not using pydantic(I doubt it)?

And frankly, I think it's overkill for type annotations longer than the actual code...

File: src/wiktextract/english_words.py [hunk 1]
```diff
@@ -6,8 +6,8 @@
 #
 # Copyright (c) 2020-2022 Tatu Ylonen.  See file LICENSE and https://ylonen.org
 
-import nltk
-from nltk.corpus import brown
+import nltk  # type: ignore[import-untyped]
```
Comment (xxyzz) [inline src/wiktextract/english_words.py:9]: Should ignore "nltk" imports in mypy config file.

**Comment by xxyzz** [comment]
And the "whitespace at end of line" ruff warning is a legit suggestion: https://docs.astral.sh/ruff/rules/blank-line-with-whitespace

**Comment by kristian-clausal** [comment]
> And the "whitespace at end of line" ruff warning is a legit suggestion: https://docs.astral.sh/ruff/rules/blank-line-with-whitespace
> 
> The [.editorconfig file](https://github.com/tatuylonen/wiktextract/blob/master/.editorconfig) was added to let the editor removes trailing whitespace.

This isn't about blank lines with whitespace, this is whitespace *at the end* of a line, which is triggered *inside quotes*, in a string. This is obviously wrong.

```
"""
This space triggers the warning -> 
"""
```

**Comment by xxyzz** [comment]
Please read the doc again...

**Comment by kristian-clausal** [comment]
You linked to W293, the error I am talking about is **W291**.

**Comment by kristian-clausal** [comment]
The ignore is also just a `# noqa:` ignore at the end of the string itself. Please read more carefully.

**Comment by xxyzz** [comment]
Both rules are created for the same reason(no trailing whitespace anywhere defined in PEP8)... If you think it's bug, you could create a bug report to the ruff project.

**Comment by kristian-clausal** [comment]
It's already been fixed in upstream ruff.

**Pull request merged**